app/helpers/sinicum/helper_utils.rb

```diff
@@ -149,16 +149,32 @@ def workspace(options)
       options[:workspace] || DEFAULT_DOCUMENT_WORKSPACE
     end
 
-    #srcset optimization, see http://w3c.github.io/html/semantics-embedded-content.html#element-attrdef-img-srcset
-    def add_srcset(attributes_hash)
+    def original_picture_width(src_attribute)
+      src_splited_path = src_attribute.split("/")
+      renderer = Sinicum::Imaging::Config.read_configuration.renderer["#{src_splited_path[2]}"]
+      renderer["x"]
+    end
+
+    def calculate_resulting_width(srcset_option, original_picture_width)
+      srcset_option = srcset_option[0...-1]
+      result = original_picture_width*srcset_option.to_f
+      result.to_i
+    end
+
+    def configure_for_srcset(attributes_hash)
       srcset_options = Sinicum::Imaging::Config.read_configuration.srcset_options
       if srcset_options.present?
         if match = attributes_hash[:src].match(/(.+?)-(\h{32})(\.\w+)?$/)
           attributes_hash[:srcset] = srcset_options.map { |e|  \
-            "#{match[1]}_#{e[0]}-#{match[2]}#{match[3]} #{e[1]}"}.join(", ")
+            "#{match[1]}_#{e[0]}-#{match[2]}#{match[3]} #{calculate_resulting_width(e[1],
+            original_picture_width(attributes_hash[:src]))}w"}.join(", ")
         end
+        #remove now unnecessary tags
+        attributes_hash.delete(:height)
```

Comment:
I'm not sure how this will work with browsers, that don't support Source Set. Please make sure, that this will not break old browsers.